pkg/core/fs_rw_ops.go

```diff
@@ -536,10 +540,12 @@ func (fs *fsMutable) FlushFile(
 	f := fs.backingFiles[op.Inode]
```

Comment:
DISCLAIMER: I am discovering this piece, so this might not be a comment for this PR, but rather a marker for future investigation.

This map access just makes our fs not concurrency-safe.

Reply:
Re concurrency-safety on ro FS: reading further, I realize that the populateFS procedure is never parallelized atm. Hence false alarm here.

Re concurrency-safety on rw FS: similarly, I inspected further how the global single lock prevents  concurrent map access to the readDir map, but not to the backingFiles map.
I am proposing a fix for that in #358